device.c

```diff
@@ -316,25 +324,42 @@ int adev_open_output_stream(struct audio_hw_device *dev,
         return -EINVAL;
     }
     /* check requested device */
-    /* TODO we should handle only devices described in XML configuration and "default" one */
     if ((devices & AUDIO_DEVICE_BIT_IN) != 0) {
         ALOGE("Failed. Incorrect device type. -EINVAL");
         *stream_out = NULL;
         pthread_mutex_unlock(&xdev->lock);
         return -EINVAL;
     }
 
-    /* TODO define card_id and dev_id for stream, using provided parameters:
-       devices, flags, address, source */
+    /* For now we have one card, so card id is always constant */
+    /* Get device id. We expect (same as parser in car audio service)
+     * that address has format "busX_%s" */
+    if (sscanf(address, "bus%u", &dev_id) != 1) {
```

Comment:
if address is a malformed, e.g. it is bus7bla, then sscanf above will return 7 as dev_id, but ignore "bla".
Do you need to check that address is exactly "bus7" in the example above, not "bus7bla?

Reply:
additional text is expected after number, so comment was expanded to provide additional info.